src/pytorch_lightning/utilities/cloud_io.py

```diff
@@ -24,6 +27,30 @@
 
 from pytorch_lightning.utilities.types import _PATH
 
+_state_observer_lock = threading.Lock()
```

Comment:
This has nothing to do with the state observer lock in the app. Can we remove it, or rename it to something else?